identity-service/src/routes/trackListens.js

```diff
@@ -116,6 +116,87 @@ const getTrackListens = async (
   return output
 }
 
+const getTrendingTracks = async (
+  idList,
+  timeFrame,
+  limit,
+  offset) => {
+  if (idList !== undefined && !Array.isArray(idList)) {
+    return errorResponseBadRequest('Invalid id list provided. Please provide an array of track IDs')
+  }
+
+  let dbQuery = {
+    attributes: ['trackId', [models.Sequelize.fn('sum', models.Sequelize.col('listens')), 'listens']],
+    group: ['trackId'],
+    order: [[models.Sequelize.col('listens'), 'DESC'], [models.Sequelize.col('trackId'), 'DESC']],
+    where: {}
+  }
+
+  // If id list present, add filter
+  if (idList && idList.length > 0) {
+    dbQuery.where.trackId = { [models.Sequelize.Op.in]: idList }
+  }
+
+  let currentHour = await getListenHour()
+  switch (timeFrame) {
+    case 'day':
+      let oneDayBefore = new Date(currentHour.getTime() - oneDayInMs)
+      dbQuery.where.hour = { [models.Sequelize.Op.gte]: oneDayBefore }
+      break
+    case 'week':
+      let oneWeekBefore = new Date(currentHour.getTime() - oneWeekInMs)
+      dbQuery.where.hour = { [models.Sequelize.Op.gte]: oneWeekBefore }
+      break
+    case 'month':
+      let oneMonthBefore = new Date(currentHour.getTime() - oneMonthInMs)
+      dbQuery.where.hour = { [models.Sequelize.Op.gte]: oneMonthBefore }
+      break
+    case 'year':
+      let oneYearBefore = new Date(currentHour.getTime() - oneYearInMs)
+      dbQuery.where.hour = { [models.Sequelize.Op.gte]: oneYearBefore }
+      break
+    case undefined:
+      break
+    default:
+      return errorResponseBadRequest('Invalid time parameter provided, use day/week/month/year or no parameter')
+  }
+  if (limit) {
+    dbQuery.limit = limit
+  }
+
+  if (offset) {
+    dbQuery.offset = offset
```

Comment:
feels pretty dangerous to not enforce any default limit/offset - esp since these routes are fully unpermissioned, i think we should add a default, min, max similar to discprov

Reply:
yes, good catch here